activerecord/lib/active_record/persistence.rb

```diff
@@ -211,12 +217,15 @@ def upsert(attributes, returning: nil, unique_by: nil)
       # ==== Options
       #
       # [:returning]
-      #   (Postgres-only) An array of attributes that should be returned for all successfully
+      #   (Postgres-only) Specify the attributes that should be returned for all successfully
       #   inserted records. For databases that support <tt>INSERT ... RETURNING</tt>, this will default
       #   to returning the primary keys of the successfully inserted records. Pass
       #   <tt>returning: %w[ id name ]</tt> to return the id and name of every successfully inserted
       #   record or pass <tt>returning: false</tt> to omit the clause.
       #
+      #   You can also pass a string if you need more control on the return values
+      #   (for example, <tt>returning: "id, name as new_name"</tt>).
```

Comment:
Should we say `<tt>returning: Arel.sql("id, name as new_name")</tt>`?

Reply:
🤔 Maybe it doesn't matter — or maybe we should `raise` if we're given an un-whitelisted String

Reply:
I think it should behave the same as another methods allowing sql string. `Post.joins('INNER JOIN posts p on 1 = 1')` works on latest master.

Forcing `Arel.sql` to whitelist sql fragements in various methods should be done in consistent way all across whole ActiveRecord API IMHO.

Reply:
I was thinking about whitelisting (`Arel.sql`) [initially](https://github.com/rails/rails/pull/35077#issuecomment-471695480) but while writing the implementation decided that it's unnecessary: as @simi mentioned, we have other methods accepting SQL strings, so, we do not force users to explicit wrapping everywhere; as far as I understand, the reason why we force `Arel.sql` in `.order` is that there is a high chance that it's used with user input (e.g. `.order("name #{params[:sort_order]}")`) and we want developers to be explicit in such _dangerous_ cases.
On the other hand, both `returning:` and `update_sql`: are unlikely to be used with _dynamic_ and user-generated string, so, there is no need in whitelisting.